src/main/java/com/example/bookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,38 @@
+package com.example.bookstore.repository.book;
+
+import com.example.bookstore.dto.BookSearchParametersDto;
+import com.example.bookstore.model.Book;
+import com.example.bookstore.repository.SpecificationBuilder;
+import com.example.bookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    @Autowired
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParameters) {
+        Specification<Book> bookSpecification = Specification.where(null);
+        if (searchParameters.authors() != null && searchParameters.authors().length > 0) {
+            bookSpecification = bookSpecification
+                    .and(bookSpecificationProviderManager.getSpecificationProvider("author")
```

Comment:
use constant for parameters. Better make constant public and reuse it in provider (the same for other)